src/Sentry/IScopeObserver.cs

```diff
@@ -0,0 +1,54 @@
+using System.Collections.Generic;
+
+namespace Sentry
+{
+    /// <summary>
+    /// Observer for the sync. of Scopes across SDKs.
+    /// </summary>
+    public interface IScopeObserver
+    {
+        /// <summary>
+        /// A trail of events which happened prior to an issue.
+        /// </summary>
+        /// <seealso href="https://docs.sentry.io/platforms/dotnet/enriching-events/breadcrumbs/"/>
+        IReadOnlyCollection<Breadcrumb> Breadcrumbs { get; }
+
+        /// <summary>
+        /// Arbitrary key-value for this event.
+        /// </summary>
+        IReadOnlyDictionary<string, string> Tags { get; }
+
+        /// <summary>
+        /// An arbitrary mapping of additional metadata to store with the event.
+        /// </summary>
+        IReadOnlyDictionary<string, object?> Extra { get; }
```

Comment:
I believe this interface should have only the methods to handle scope change and not expose any state
```suggestion
```